### PR TITLE
TEP-0114: Switch to use FilterCustomRunRef - Wait Custom Task Beta

### PR DESCRIPTION
**Pull request by XinruZhang**
# Changes

Wait Custom Task Beta controller was using an inline filter func when adding event handler to the customruninformer, pr https://github.com/tektoncd/pipeline/pull/5822 splitted the inline func out, and this commit uses the refactored, shared filter func instead in the Wait Custom Task Beta controller.

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: test/custom-task-ctrls/wait-task-beta/go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.15
 require (
 	github.com/emicklei/go-restful v2.15.0+incompatible // indirect
 	github.com/google/go-cmp v0.5.9
-	github.com/tektoncd/pipeline v0.42.0
+	github.com/tektoncd/pipeline v0.43.2
```
Comment (JeromeJu) [inline test/custom-task-ctrls/wait-task-beta/go.mod:8]: Wondering if this shall be put in the release note?

**Review by JeromeJu — APPROVED** [review]
Thanks for the cleanup @XinruZhang !

File: test/custom-task-ctrls/wait-task-beta/go.mod (hunk 1 above)
Comment (XinruZhang) [reply test/custom-task-ctrls/wait-task-beta/go.mod:8]: This is a controller used interally for integration test, it's not user facing :)

File: test/custom-task-ctrls/wait-task-beta/pkg/reconciler/reconciler_test.go [hunk 2]
```diff
@@ -397,20 +396,14 @@ func TestReconcile(t *testing.T) {
 	}
 }
 
-// TODO: Delete when we switch its usage to parse.MustParseCustomRun once that change is in a release.
```
[the remark this answers is not in the document]
Comment (XinruZhang) [reply test/custom-task-ctrls/wait-task-beta/pkg/reconciler/reconciler_test.go:400]: If you search for "MustParseCustomRun", it's already in use :)

**Pull request merged**